aten/src/ATen/native/ResizeCommon.h

```diff
@@ -6,16 +6,28 @@
 namespace at { namespace native {
 
 #ifdef BUILD_NAMEDTENSOR
-inline Tensor& resize_named_tensor_(Tensor& self, IntArrayRef size) {
+inline Tensor& resize_named_tensor_(
+    Tensor& self,
+    IntArrayRef size,
+    c10::optional<MemoryFormat> optional_memory_format) {
   TORCH_INTERNAL_ASSERT(self.has_names());
   TORCH_CHECK(
       self.sizes() == size,
       "Cannot resize named tensor with resize_ or resize_as_ (tried to resize "
-      "Tensor", self.names(), " with size ", self.sizes(), " to ", size,
+      "Tensor",
+      self.names(),
+      " with size ",
+      self.sizes(),
+      " to ",
+      size,
       "). This may be caused by passing a named tensor ",
       "as an `out=` argument; please ensure that the sizes are the same. ");
+  TORCH_CHECK(
+      optional_memory_format.value_or(MemoryFormat::Contiguous) ==
```

Comment:
shouldn't this just checking if optional_memory_format has a value?

The behavior is we only restride if there's a value, and here we don't restride.